de.dlr.sc.virsat.model.extension.tests.edit.test/src/de/dlr/sc/virsat/model/dvlm/dmf/DmfResourceTest.java

```diff
@@ -659,8 +660,85 @@ protected void doExecute() {
 		TestCategoryReference beanTestCategoryAssignmentReference = new TestCategoryReference(referencingCa);
 		
 		assertEquals("Reference is set correctly", beanTestCategoryAssignment, beanTestCategoryAssignmentReference.getTestRefCategory());
+		
+		resSeiOther.unload();
+		resSei.unload();
+		dmfCategoryAssignmentReference.setName("vfdv");
+		dmfCategoryAssignment.setName("sdfsdf");
+		dmfResource.save(Collections.EMPTY_MAP);
+		
 	}
 	
+	@Test
+	public void testSaveReferencedDObjectDifferentUnloadedContainment() throws IOException {
+		TestStructuralElement beanTestStructuralElement = new TestStructuralElement(concept);
+		TestStructuralElement beanTestStructuralElementOther = new TestStructuralElement(concept);
+
+		ed.getVirSatCommandStack().execute(new RecordingCommand(ed) {
+			@Override
+			protected void doExecute() {
+				resSet.getAndAddStructuralElementInstanceResource(beanTestStructuralElement.getStructuralElementInstance());
+				resSet.getAndAddStructuralElementInstanceResource(beanTestStructuralElementOther.getStructuralElementInstance());
+			}
+		});
+
+		Resource resSei = resSet.getStructuralElementInstanceResource(beanTestStructuralElement.getStructuralElementInstance());
+		Resource resSeiOther = resSet.getStructuralElementInstanceResource(beanTestStructuralElementOther.getStructuralElementInstance());
+		
+		Resource.Factory.Registry resourceRegistry = Resource.Factory.Registry.INSTANCE;
+		Map<String, Object> m = resourceRegistry.getExtensionToFactoryMap();
+		m.put(DmfResource.DMF_FILENAME_EXTENSION, new DmfResourceFactory());
+
+		URI originalSeiUriOther = resSeiOther.getURI();
+		URI dmfSeiUriOther = originalSeiUriOther.appendFileExtension(DmfResource.DMF_FILENAME_EXTENSION);
+		
+		URI originalSeiUri = resSei.getURI();
+		URI dmfSeiUri = originalSeiUri.appendFileExtension(DmfResource.DMF_FILENAME_EXTENSION);
+		
+		ResourceSet dmfResourceSet = new ResourceSetImpl();
+		Resource dmfResourceOther = dmfResourceSet.getResource(dmfSeiUriOther, true);
+		Resource dmfResource = dmfResourceSet.getResource(dmfSeiUri, true);
+		
+		ed.saveAll();
+		
+		DObjectContainer dObjectContainerOther = (DObjectContainer) dmfResourceOther.getContents().get(0);
+		de.dlr.sc.virsat.model.extension.tests.tests.TestCategoryReference dmfCategoryAssignmentReference = TestsFactory.eINSTANCE.createTestCategoryReference();
+		de.dlr.sc.virsat.model.extension.tests.tests.TestCategoryAllProperty dmfCategoryAssignment = TestsFactory.eINSTANCE.createTestCategoryAllProperty();
+
+		dObjectContainerOther.getObjects().add(dmfCategoryAssignment);
+		dmfResourceOther.save(Collections.EMPTY_MAP);
+		
+		DObjectContainer dObjectContainer = (DObjectContainer) dmfResource.getContents().get(0);
+		dmfCategoryAssignmentReference.setTestRefCategory(dmfCategoryAssignment);
+		dObjectContainer.getObjects().add(dmfCategoryAssignmentReference);
+		dmfResource.save(Collections.EMPTY_MAP);
+		
+		//Unload original DVLM resources

```

Comment:
Please add explanation why to undload the DVLMs. What is the expected bahvior. And what should happen in the backgrund?